src/llvm-late-gc-lowering.cpp

```diff
@@ -2306,6 +2319,15 @@ void LateLowerGCFrame::PlaceGCFrameStores(State &S, unsigned MinColorRoot,
         for(auto rit = BBS.Safepoints.rbegin();
               rit != BBS.Safepoints.rend(); ++rit ) {
             const LargeSparseBitVector &NowLive = S.LiveSets[*rit];
+            // reset slots which are no longer alive
+            for (int Idx : *LastLive) {
+                if (Idx >= PreAssignedColors && !HasBitSet(NowLive, Idx)) {
+                    PlaceGCFrameReset(S, Idx, MinColorRoot, Colors, GCFrame,
```

Comment:
Would it be better to compute which slot colors transition? LLVM can merge the stores, but it's a bit of an expensive computation for it to do that.

Reply:
That might be interesting. Especially since we we would know if skipped slots are already null (and thus can be written with a larger store), but maybe as a later followup. I think our bigger problem right now is how exhaustively double-rooted everything is due to lack of any stack coloring

Reply:
Sounds good to me.